### PR TITLE
@View to @BaseView rename

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
refactor(@View): rename @View to @BaseView

BREAKING CHANGE:

@View was renamed to @BaseView

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
Hmm, normally this would be good to go but there are some Dart transformer tests failing and I it is not obvious for me why :-/ Will continue to poke around but if anyone familiar with Dart transformers could give me a hand (@kegluneq  ?), this would be awesome.

**Comment by pkozlowski-opensource** [comment]
So, I've managed to make the TravisCI green, but it still needs solid review

**Comment by pkozlowski-opensource** [comment]
Rebased on top of the current maser. I would love to get this in sooner than later, as the diff is not trivial

**Comment by kegluneq** [comment]
Sorry for the delay - I've been out of office dealing with a house move gone wrong these past few days.

LGTM for Dart transformer changes, but I am almost certainly not the correct person for a full review.

File: modules/angular2/directives.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export * from './src/directives/ng_switch';
  * @Component({
  *  selector: 'my-component'
  * })
- * @View({
+ * @BaseView({
```
Comment (mhevery) [inline modules/angular2/directives.ts]: I am not sure if we should change all of the docs to show `BaseView` What do you think?

File: modules/angular2/src/core/annotations/decorators.ts [hunk 2]
```diff
@@ -219,11 +219,11 @@ export interface ComponentFactory {
  *
  * MyComponent.annotations = [
  *   new ng.Component({...})
- *   new ng.View({...})
+ *   new ng.BaseView({...})
  * ]
  * ```
  */
-export interface ViewFactory {
+export interface BaseViewFactory {
```
Comment (mhevery) [inline modules/angular2/src/core/annotations/decorators.ts]: I think this should stay `ViewFactory` The `Base` is confusing I think.

File: modules/angular2/src/core/annotations/decorators.ts [hunk 3]
```diff
@@ -239,7 +239,7 @@ export interface ViewFactory {
     encapsulation?: ViewEncapsulation,
     styles?: List<string>,
     styleUrls?: List<string>,
-  }): ViewAnnotation;
+  }): BaseViewAnnotation;
```
Comment (mhevery) [inline modules/angular2/src/core/annotations/decorators.ts]: Same here?

**Comment by mhevery** [comment]
Per discussion with @pkozlowski-opensource we are closing this PR

Step forward:

`metadata.ts`:
  - introduce a `BaseView` so that we can do something like this.
    `var View = BaseView.extend({ directives: CORE_DIRECTIVES });`
- `metadat.dart`:
  - introduce a `BaseView` so that class hierarchy is: 
    `View extends BaseView extends ViewMetadata`

In all internal directives (`ng-if`, etc) use `@BaseView` instead of `View`.

No additional work/renaming is required.

**Pull request closed without merging**